sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -673,55 +721,72 @@ def plot(self, ax=None, n_cols=3, line_kw=None, contour_kw=None):
                 preds = pd_result.individual
 
             if len(values) == 1:
-                if self.kind == 'individual' or self.kind == 'both':
-                    n_samples = self._get_sample_count(
-                        len(preds[self.target_idx])
-                    )
+                if self.kind in ("individual", "both"):
                     ice_lines = preds[self.target_idx]
-                    sampled = ice_lines[np.random.choice(
-                        ice_lines.shape[0], n_samples, replace=False
-                    ), :]
-                    for j, ins in enumerate(sampled):
-                        lines_ravel[i * j + j] = axi.plot(
-                            values[0], ins.ravel(), **individual_line_kw
-                        )[0]
-                if self.kind == 'average':
-                    lines_ravel[i] = axi.plot(
-                        values[0], avg_preds[self.target_idx].ravel(),
-                        **line_kw
-                    )[0]
-                elif self.kind == 'both':
-                    lines_ravel[i] = axi.plot(
-                        values[0], avg_preds[self.target_idx].ravel(),
-                        label='average', **line_kw
+                    ice_lines_idx = rng.choice(
+                        ice_lines.shape[0], n_ice_to_plot, replace=False
+                    )
+                    ice_lines_subsampled = ice_lines[ice_lines_idx, :]
+                    for ice_idx, ice in enumerate(ice_lines_subsampled):
+                        lines_ravel[
+                            pd_plot_idx * n_lines + ice_idx
+                        ] = axi.plot(
+                            values[0], ice.ravel(), **individual_line_kw
+                        )[
+                            0
+                        ]
```

Comment:
Is this... how black formats this?

Reply:
Yep. Black is a great tool , but some design decisions are... special... In my opinion something like the following lines would be much better.
```
                        ] = axi.plot(
                              values[0], ice.ravel(), **individual_line_kw
                        )[0]
```

Reply:
> Is this... how black formats this?

I let it do :) and hoping for the best